### PR TITLE
Acceptance test for favorites page

### PR DESCRIPTION
**Pull request by skshetry**
## Description

This adds a test scenario for favoriting files and making sure that they exist in the favorites page. More scenario are needed. To accustom for the full path shown in the page, `fileList()` had to be changed, which now checks for whole filename instead of an attribute (which is a good thing?).

## Related Issue

- Related to #1194

## How Has This Been Tested?

- locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: tests/acceptance/pageObjects/FilesPageElement/filesList.js [hunk 1]
```diff
@@ -166,11 +169,20 @@ module.exports = {
...
-      var element = this.elements['fileRowByName']
-      var util = require('util')
+      // replace double quotes if there any to single quotes
       if (fileName.indexOf('"') > -1) {
-        element.selector = element.selector.replace(/"/g, "'")
+        fileName = fileName.replace(/"/g, "'")
+      }
+
+      const parts = path.parse(fileName)
+      if (parts.ext) {
```
Comment (skshetry) [inline tests/acceptance/pageObjects/FilesPageElement/filesList.js]: Currently, we have a bug that shows the link for the folder having a period splitted (ie. without extension). If this gets fixed, we might need to use regex step definition to pass if it's really a folder or a file. This way, we could avoid splitting if it's a folder.

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIRenameFolders</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/3717/9

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFiles</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/3717/5

<p>





</p></details>

File: tests/acceptance/stepDefinitions/filesContext.js [hunk 2]
```diff
@@ -148,6 +148,10 @@ Then('file/folder {string} should not be listed on the webUI', function (folder)
   return client.page.FilesPageElement.filesList().assertElementNotListed(folder)
 })
 
+Then('file/folder with name {string} should be listed in the favorites page on the webUI', function (name) {
```
Comment (paurakhsharma) [inline tests/acceptance/stepDefinitions/filesContext.js]: We don't need to specify favorites page here.

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/3717/23

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIRenameFiles</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/3717/8

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/3717/24

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIRenameFolders</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/3719/10

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/3719/24

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIRenameFiles</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/3719/9

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFiles</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/3719/5

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/3719/25

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIRenameFolders</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/3742/10

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIRenameFiles</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/3742/9

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/3742/24

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFiles</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/3742/5

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/3742/25

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/3763/25

<p>





</p></details>

File: tests/acceptance/features/webUIFavorites/favoritesFile.feature [hunk 3]
```diff
@@ -69,8 +67,9 @@ Feature: Mark file as favorite
...
     And the user opens folder "strängé नेपाली folder" using the webUI
     And the user marks file "lorem.txt" as favorite using the webUI
-    Then file "lorem.txt" with path "/" should be listed in the favorites page on the webUI
-    And file "lorem.txt" with path "/simple-folder" should be listed in the favorites page on the webUI
-    And folder "simple-empty-folder" with path "/" should be listed in the favorites page on the webUI
-    And file "simple-empty-folder" with path "/simple-folder" should be listed in the favorites page on the webUI
-    And file "lorem.txt" with path "/strängé नेपाली folder" should be listed in the favorites page on the webUI
+    When the user browses to the favorites page
+    Then file "lorem.txt" should be listed on the webUI
+    Then file "simple-folder/lorem.txt" should be listed on the webUI
```
Comment (paurakhsharma) [inline tests/acceptance/features/webUIFavorites/favoritesFile.feature]:
```suggestion
    And file "simple-folder/lorem.txt" should be listed on the webUI
```

File: tests/acceptance/features/webUIFiles/createFolderEdgeCases.feature [hunk 4]
```diff
@@ -18,6 +18,7 @@ Feature: create folder
       | 'सिमप्ले फोल्देर $%#?&@' |
       | '"somequotes1"'          |
       | "'somequotes2'"          |
+      | "\"quote\"d-folders'"    |
```
Comment (individual-it) [inline tests/acceptance/features/webUIFiles/createFolderEdgeCases.feature]: why this change?

Comment (skshetry) [reply tests/acceptance/features/webUIFiles/createFolderEdgeCases.feature]: Just wanted to make sure the `concat` part is doing it's job. I know it's bad to test for the test code itself, but it was a change in a critical part of the test code and a valid requirement.

Comment (individual-it) [reply tests/acceptance/features/webUIFiles/createFolderEdgeCases.feature]: Please make then a separate commit for the `concat` part

Comment (skshetry) [reply tests/acceptance/features/webUIFiles/createFolderEdgeCases.feature]: Done. Created separate PR for this feature file addition.

File: tests/acceptance/helpers/xpath.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+   * as xpath expressions can contain single or double quotes but not both.
+   * Also, xpath does not support escaping.
+   *
+   * {@link https://stackoverflow.com/q/1341847 Special Character in XPATH Query}
+   * {@link https://stackoverflow.com/q/642125 Encoding XPath Expressions with both single and double quotes}
+   *
+   * @param {string} value
+   * @returns {string}
+   */
+  buildXpathLiteral: function (value) {
```
Comment (skshetry) [inline tests/acceptance/helpers/xpath.js]: I tested this function with following folders as well:
```
        "foo",              // no quotes
        "\"foo",            // double quotes only
        "'foo",             // single quotes only
        "'foo\"bar",        // both; double quotes in mid-string
        "'foo\"bar\"baz",   // multiple double quotes in mid-string
        "'foo\"",           // string ends with double quotes
        "'foo\"\"",         // string ends with run of double quotes
        "\"'foo",           // string begins with double quotes
        "\"\"'foo",         // string begins with run of double quotes
        "'foo\"\"bar"       // run of double quotes in mid-string
```
